File: 251051/Manukh/KR_1_REST/src/main/java/com/example/repository/CrudRepository.java
// CrudRepository.java
package com.example.repository;

import com.example.model.BaseEntity;

import java.util.List;
import java.util.Optional;

public interface CrudRepository<T extends BaseEntity, K> {
    List<T> findAll();
    Optional<T> findById(K id);
    T save(T entity);
    T update(T entity);
    boolean deleteById(K id);
    boolean existsById(K id);
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/repository/CrudRepository.kt
package com.example.distcomp.repository

import com.example.distcomp.model.BaseEntity

interface CrudRepository<T : BaseEntity, ID> {
    fun save(entity: T): T
    fun findById(id: ID): T?
    fun findAll(): List<T>
    fun deleteById(id: ID): Boolean
    fun existsById(id: ID): Boolean
}
